libs/wingsdk/src/shared/bundling.ts

```diff
@@ -101,3 +94,100 @@ export function createBundle(
     sourcemapPath: outfileMap,
   };
 }
+
+interface SourceMap {
+  sourceRoot: string;
+  sources: string[];
+  sourcesContent: string[];
+  mappings: string;
+}
+
+/**
+ * Takes a bundled sourcemap and does the following fixes:
+ * - Normalizes paths in sources and sourceRoot
+ * - Removes duplicate sources and sourcesContent
+ * - Updates mappings to reflect the new source indices
+ *
+ * The duplicate sources come from esbuild's strange handling of multiple files being bundled that point to the same source (e.g. inflights that point to one .w file)
+ * See https://github.com/evanw/esbuild/issues/933
+ */
+function fixSourcemaps(sourcemapData: SourceMap): void {
```

Comment:
if/when we do async esbuild, this would probably be much more efficient as a plugin